module/os/linux/spl/spl-thread.c

```diff
@@ -158,3 +158,37 @@ spl_kthread_create(int (*func)(void *), void *data, const char namefmt[], ...)
 	} while (1);
 }
 EXPORT_SYMBOL(spl_kthread_create);
+
+/*
+ * The "why" argument indicates the allowable side-effects of the call:
+ *
+ * FORREAL:  Extract the next pending signal from p_sig into p_cursig;
+ * stop the process if a stop has been requested or if a traced signal
+ * is pending.
+ *
+ * JUSTLOOKING:  Don't stop the process, just indicate whether or not
+ * a signal might be pending (FORREAL is needed to tell for sure).
+ */
+int
+issig(int why)
+{
+	ASSERT(why == FORREAL || why == JUSTLOOKING);
+
+	if (!signal_pending(current))
+		return (0);
+
+	if (why == FORREAL &&
+	    (sigismember(&current->signal->shared_pending.signal, SIGSTOP) ||
+	    sigismember(&current->signal->shared_pending.signal, SIGTSTP))) {
+		/* TODO: right semantics? */
+		int sig = kernel_dequeue_signal();
```

Comment:
It appears neither `kernel_dequeue_signal()` or `kernel_signal_stop()` are available in older kernels.  For these kernels it looks like you'll need a new configure and some compatibility code to do it the old way.  You should be able to use `dequeue_signal_lock()`, this function was renamed  `kernel_dequeue_signal()` for newer kernels and reworked slightly.  You should also be able add your own helper function for `kernel_signal_stop()` which takes the `tsk->sighand->siglock` and sets the state to `TASK_STOPPED` with `set_current_state()`.